### PR TITLE
history view follows preferences

### PR DESCRIPTION
**Pull request by mbkma**
Test:
enter a few big numbers
In history view, the result adapts to current settings like Number Format and Decimal Places.

### CODE REVIEW

File: src/math-history.c [hunk 1]
```diff
@@ -54,40 +54,25 @@ scroll_bottom_cb(MathHistory *history, gpointer data)
 void
 math_history_insert_entry (MathHistory *history, char *equation, MPNumber *answer, int number_base)
 {
```
Comment (raveit65) [inline src/math-history.c:56]: There is a new build warning.
```
math-history.c:55:88: warning: unused parameter 'number_base' [-Wunused-parameter]

math_history_insert_entry (MathHistory *history, char *equation, MPNumber *answer, int number_base)
```
https://travis-ci.org/github/mate-desktop/mate-calc/jobs/759745686#L1704

**Comment by raveit65** [comment]
Only rebased with master to eliminated tha archlinux build issue.

**Comment by raveit65** [comment]
@rbuj @lukefromdc 
This is a huge commit, does it work for you?

**Review by rbuj — APPROVED** [review]
LGTM.

Test:
- Set Programming mode (Mode -> Programming)
- Enable history (View -> History)
- Set octal number system
- Write 17
- Set hexadecimal number system

The history is updated and it displays F

**Review by lukefromdc — APPROVED** [review]
Builds and runs fine, all the suggested tests worked

**Pull request merged**